### ISSUE TITLE
runtime: `lock ordering problem` between `pollDesc` and `prof`

### ISSUE THREAD

**Issue by bcmills** [golang/go#57637]
```
#!watchflakes
post <- `lock ordering problem` && `^\d+ : pollDesc ` && `\d+ : prof `
```

From https://github.com/golang/go/issues/55167#issuecomment-1371429539:

```
60879  ======
0 : pollDesc 8 0x7fdcf0642a90
1 : prof 22 0x84c5c0
fatal error: lock ordering problem
```

**Comment by bcmills** [comment]
(attn @golang/runtime)

**Comment by prattmic** [comment]
Note that this failure is on release-branch.go1.18.

I believe this was fixed by https://go.dev/cl/418715. `pollDesc < prof` was already an allowed order, but the direct edge was not listed in the partial order. https://go.dev/cl/418715 adds it to the partial order.

I could make a minimal backport if desired, but given this has only triggered once, I'm not sure it is worth it.

**Comment by bcmills** [comment]
Given the relatively short remaining shelf-life of Go 1.18, I agree that this probably isn't worth backporting. Thanks for looking into it!

**Comment by prattmic** [comment]
This bug affects 1.19 as well, not sure if that changes your thoughts.

**Comment by bcmills** [comment]
Hmm. We're going to have 1.19 around for a whole release cycle (so it might be worth fixing to avoid spurious failures during patch releases), but this failure mode is also rare (end users probably _never_ see it, because they don't build with `GOEXPERIMENT=staticlockranking`).

So, I'm personally ambivalent about backporting. 🤷‍♂️

**Comment by aclements** [comment]
I don't consider `GOEXPERIMENT=staticlockranking` to be a user-facing build mode, and given that this was merely a missing edge in the old lock graph and not an actual order violation, I don't think we should bother with a fix to 1.19 unless the failures are bothering *us*.

**Comment by cherrymui** [comment]
Sounds like we decide to not backport. Closing for now. If it actually bothers us we can reopen and backport.
